examples/hello/client.c

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (C) Bull S.A.S - 2024
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program; if
+ * not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ *
+ * BXI Low Level Team
+ *
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "portals4.h"
+#include "portals4_bxiext.h"
+
+#include "hello.h"
+
+/*
+ * this file represent a client sendind a message to a server
+ */
+
+int main(int argc, char **argv)
+{
+	int res;
+	int ret;
+	ptl_handle_ni_t nih;
+	ptl_handle_eq_t eqh;
+	ptl_handle_md_t mdh;
+	ptl_event_t ev;
+	ptl_hdr_data_t header_data = 1234;
+	ptl_process_t id_server = { .phys = { .pid = SERVER_PID, .nid = 0 } };
+	long nid;
+	char msg[PTL_EV_STR_SIZE];
+	char *p;
+	/* represent the memory used to send the message */
+	ptl_md_t md = { .start = argv[2],
+			.length = strlen(argv[2]) + 1,

```

Comment:
The strlen must be called after checking the number of arguments, else it will crash if the message is not supplied